content/articles/multiclass-text-classification-with-pyspark/index.md

```diff
@@ -0,0 +1,730 @@
+5. Download the Udemy dataset [here](https://drive.google.com/file/d/1-TfybHeSWpP9UJLxeNDIFT1CPfxn7XWT/view?usp=sharing).
+
+> NOTE: To follow along easily, use [Jupyter Notebook](https://jupyter.org/) to build your text classification model.
+
+### Introduction
+Pyspark uses the Spark API in data processing and model building. Spark API consists of the following libraries:
+
+1. Spark SQl.
+2. Spark Streaming.
+3. MLib.
+4. Spark Core.
+5. GraphX.
+
+#### [Spark SQL](https://spark.apache.org/docs/latest/api/python/)
+
+This is the structured query language used in the data processing. It's used to query the datasets in exploring the data used in model building.
+
+#### Spark Streaming
+
+Allows processing and analysis of real-time data from various sources such as [Flume](https://flume.apache.org/), [Kafka](https://kafka.apache.org/) and [Amazon Kinesis.](https://aws.amazon.com/kinesis/)
+
+The image below shows the components of spark streaming:
+
+![Spark Streaming](/engineering-education/multiclass-text-classification-with-pyspark/spark-streaming.png)
+
+*[Image source: Databricks](https://databricks.com/wp-content/uploads/2018/05/Apache-Spark-Streaming-ecosystem-diagram.png)*
+
+#### [MLib](https://spark.apache.org/mllib/)
+
+Contains a uniform set of high-level APIs used in model creation. Help to train our model and find the best algorithm.
+
+#### [Spark Core](https://spark.apache.org/docs/latest/api/python/)
+
+This is the root of the Spark API. It's involved with the core functionalities such as basic I/O functionalities, task scheduling, and memory management.
+
+#### [GraphX](https://spark.apache.org/docs/latest/graphx-programming-guide.html)
+
+Used in plotting of graphs for Spark computations.
+
+The image below shows components of the Spark API:
+
+![Components-of-Spark-API](/engineering-education/multiclass-text-classification-with-pyspark/components-of-spark.jpg)
+
+*[Image source: Tutorialspoint](https://www.tutorialspoint.com/apache_spark/images/components_of_spark.jpg)*
+
+Pyspark supports two data structures that are used during data processing and machine learning building:
+
+1. Resilient Distributed Dataset(RDD).
+2. Dataframe.
+
+#### [Resilient Distributed Dataset(RDD)](https://www.tutorialspoint.com/apache_spark/apache_spark_core_programming.htm)
+
+This is a distributed collection of data spread and distributed across multiple machines in a cluster.
+
+RDD is best described in three ways.
+
+- Resilient: Fault-tolerant and can rebuild itself if a failure occurs.
+- Distributed: Data is distributed and spread across multiple machines in a cluster.
+- Dataset: Collection of partitioned data with values.
+
+#### [Dataframe](https://www.tutorialspoint.com/apache_spark/apache_spark_core_programming.htm)
+
+This is distributed collection of either structured or semi-structured datasets. Dataset has organized columns in relational databases and excels sheets and its current dataset structure.
```

Comment:
This sentence isn't clear

`Dataset has organized columns in relational databases and excels sheets and its current dataset structure.`